Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM rust:1.74.0-alpine3.18 as builder
+
+WORKDIR /build
+
+RUN apk add --no-cache musl-dev=1.2.4-r2
+
+COPY .cargo .
+COPY Cargo* ./
+COPY src src/
+
+RUN cargo build --release --locked --target=x86_64-unknown-linux-musl
```

Comment:
I think you can cache the dependencies with a trick

Copy `Cargo*`  first, then create a dummy `lib.rs` file:

```dockerfile
RUN mkdir -p src && touch src/lib.rs && cargo build --release --locked --target=x86_64-unknown-linux-musl
```

then copy src and run the build again.
For the CI it shouldn't matter, but for local testing it would probably speed up things considerably.

Reply:
It does help.

Changed in [33320ee](https://github.com/Angelin01/pod-director/pull/16/commits/33320eea18925a42a78090fa2cdab5323c1eb54f)